lib/bathlarp_web/api_auth_plug.ex

```diff
@@ -28,31 +28,40 @@ defmodule BathLARPWeb.APIAuthPlug do
 
   The tokens are added to the `conn.private` as `:api_access_token` and
   `:api_renewal_token`. The renewal token is stored in the access token
-  metadata and vice versa.
+  metadata and vice versa. The expiry timestamp of the access token is
+  also stored as `:api_access_expiry` so it can be referenced by the
+  frontend later.
   """
   @impl true
   @spec create(Conn.t(), map(), Config.t()) :: {Conn.t(), map()}
   def create(conn, account, config) do
     store_config = store_config(config)
     access_token = Pow.UUID.generate()
     renewal_token = Pow.UUID.generate()
+    token_conf = Application.get_env(:bathlarp, BathLARPWeb.APIAuthPlug)
+    access_expiry = Keyword.get(token_conf, :access_expiry) |> String.to_integer()
+    refresh_expiry = Keyword.get(token_conf, :refresh_expiry) |> String.to_integer()
 
     conn =
       conn
       |> Conn.put_private(:api_access_token, sign_token(conn, access_token, config))
       |> Conn.put_private(:api_renewal_token, sign_token(conn, renewal_token, config))
+      |> Conn.put_private(
+        :api_access_expiry,
+        DateTime.utc_now() |> DateTime.add(access_expiry, :minute)
+      )
       |> Conn.register_before_send(fn conn ->
         # The store caches will use their default `:ttl` setting. To change the
         # `:ttl`, `Keyword.put(store_config, :ttl, :timer.minutes(10))` can be
         # passed in as the first argument instead of `store_config`.
         CredentialsCache.put(
-          store_config,
+          Keyword.put(store_config, :ttl, :timer.minutes(access_expiry)),
           access_token,
           {account, [renewal_token: renewal_token]}
         )
 
         PersistentSessionCache.put(
-          store_config,
+          Keyword.put(store_config, :ttl, :timer.minutes(refresh_expiry * 60 * 24)),
```

Comment:
suggestion: Unfortunately `:timer` doesn't have a `days` function, but we can still simplify this a little:

```suggestion
          Keyword.put(store_config, :ttl, :timer.hours(refresh_expiry * 24)),
```